src/pages/NewChatPage.tsx

```diff
@@ -158,27 +167,45 @@ function NewChatPage({betas, isGroupChat, personalDetails, reports, isSearchingF
      * or navigates to the existing chat if one with those participants already exists.
      */
     const createChat = (option: OptionData) => {
-        if (!option.login) {
+        let login = '';
+
+        if (option.login) {
+            login = option.login;
+        } else if (selectedOptions.length === 1) {
+            login = selectedOptions[0].login ?? '';
+        }
+
+        if (!login) {
+            Log.warn('Tried to create chat with empty login');
             return;
         }
-        Report.navigateToAndOpenReport([option.login]);
-    };
 
+        Report.navigateToAndOpenReport([login]);
+    };
     /**
-     * Creates a new group chat with all the selected options and the current user,
-     * or navigates to the existing chat if one with those participants already exists.
+     * Navigates to create group confirm page
      */
-    const createGroup = () => {
-        const logins = selectedOptions.map((option) => option.login).filter((login): login is string => typeof login === 'string');
-
-        if (logins.length < 1) {
+    const navigateToConfirmPage = () => {
+        if (!personalData || !personalData.login || !personalData.accountID) {
             return;
         }
-
-        Report.navigateToAndOpenReport(logins);
+        const selectedParticipants: SelectedParticipant[] = selectedOptions.map((option: OptionData) => ({login: option.login ?? '', accountID: option.accountID ?? -1}));
+        const logins = [...selectedParticipants, {login: personalData.login, accountID: personalData.accountID}];
+        Report.setGroupDraft(logins);
+        Navigation.navigate(ROUTES.NEW_CHAT_CONFIRM);
     };
 
     const updateOptions = useCallback(() => {
+        let newSelectedOptions;
+        if (newGroupDraft?.participants) {
+            const selectedParticipants = newGroupDraft.participants.filter((participant) => participant.accountID !== personalData.accountID);
+            newSelectedOptions = selectedParticipants.map((participant): OptionData => {
+                const baseOption = OptionsListUtils.getParticipantsOption({accountID: participant.accountID, login: participant.login, reportID: ''}, personalDetails);
+                return {...baseOption, reportID: baseOption.reportID ?? ''};
+            });
+            setSelectedOptions(newSelectedOptions);
+        }
```

Comment:
This overwrites the selected participants with the draft values everytime the options are re-rendered making it impossible to control the selection properly. Caused issue https://github.com/Expensify/App/issues/39295. FIx https://github.com/Expensify/App/pull/39312